### ISSUE TITLE
`prefer_self_in_static_references` doesn't report violations in extensions

### ISSUE THREAD

**Issue by rgoldberg** [realm/SwiftLint#6828]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

`prefer_self_in_static_references` doesn't report violations in extensions

```swift
struct Example {
	var value = 1
}

extension Example {
	static func example() -> Example { // no violation reported here
		.init()
	}
}
```

```bash
$ swiftlint lint --quiet --only-rule prefer_self_in_static_references Sources
```

### Environment

* SwiftLint version: 0.65.0
* Xcode version: Xcode 26.6 / Build version 17F113
* Installation method: Homebrew
* Configuration file: occurs even if no config file

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
